### PR TITLE
fix: one more Other(Cancelled) compaction error

### PR DESCRIPTION
**Pull request by koivunej**
evidence: https://neon-github-public-dev.s3.amazonaws.com/reports/pr-8655/10301613380/index.html#suites/cae012a1e6acdd9fdd8b81541972b6ce/653a33de17802bb1/

### CODE REVIEW

**Comment by koivunej** [comment]
Argh, this might not be enough, I missed that cancelled is already piped through.

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -746,7 +746,7 @@ impl From<CreateImageLayersError> for CompactionError {
             CreateImageLayersError::Other(e) => {
                 CompactionError::Other(e.context("create image layers"))
             }
-            _ => CompactionError::Other(e.into()),
+            value => CompactionError::Other(value.into()),
```
Comment (koivunej) [inline pageserver/src/tenant/timeline.rs:749]: I'll remove this accidential one as soon as the tests have finished.

**Review by arpad-m — APPROVED** [review]

**Comment by koivunej** [comment]
This was not it actually. It's not easy to reproduce, but it just happened during deploy.

**Comment by koivunej** [comment]
I think we need to mark all `Cancelled` variants with names pointing back to the type defining it so we can get rid of these once and for all.

**Comment by koivunej** [comment]
I think I've found the source finally BUT i will do it on a new PR.

**Comment by koivunej** [comment]
Superceded with #8697

**Pull request closed without merging**